### PR TITLE
Upgrade RuboCop to v1.30

### PR DESCRIPTION
**Pull request by mishina2228**
RuboCop v1.30.0 can now parse Ruby 2.0 code.
This PR upgrades RuboCop to v1.30 and also fixes and suppresses some cops.
It was hard to fix all cops, so I fixed only the simple ones.

This PR supersedes #1256.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -19,7 +19,7 @@ group :test do
   # Downgrade psych because old RuboCop can't use new Psych
   gem 'psych', '< 4.0.0'
```
Comment (sodabrew) [inline Gemfile:20]: The explicit `gem 'psych'` line can now be removed

File: Gemfile [hunk 2]
```diff
@@ -2,11 +2,11 @@ source 'https://rubygems.org'
 
 gemspec
 
-gem 'rake', if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new("2.2")
-              '~> 13.0.1'
-            else
-              '< 13'
-            end
+if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new("2.2")
```
Comment (sodabrew) [inline Gemfile:5]: I think the next release could drop Ruby < 2.2

**Review by sodabrew — APPROVED** [review]
Looks great, thanks for doing this! The only specific change I'd like is removing the extra `psych` gem that was pinned lower for older versions of RuboCop.

The other comment about older versions of Rake can be dealt with in a Ruby 2.x or higher pass. There may be some other old syntax or old C extension methods that can be cleaned up as well.

File: Gemfile (hunk 2 above)
Comment (mishina2228) [reply Gemfile:5]: So can we drop Ruby 2.1 from the CI matrix?
https://github.com/brianmario/mysql2/blob/80e0001f87ffd3974835478d2e9b068cc5d37f71/.github/workflows/build.yml#L18-L27

and also CentOS 7 with Ruby 2.0.0?
https://github.com/brianmario/mysql2/blob/80e0001f87ffd3974835478d2e9b068cc5d37f71/.github/workflows/container.yml#L14-L15

**Pull request merged**